src/main.ts

```diff
@@ -811,6 +815,127 @@ function generateDecode(ctx: Context, fullName: string, messageDesc: DescriptorP
   return joinCode(chunks, { on: '\n' });
 }
 
+function generateDecodeDelimited(ctx: Context, fullName: string, messageDesc: DescriptorProto): Code {
+  const { options, utils, typeMap } = ctx;
+  const chunks: Code[] = [];
+
+  // create the basic function declaration
+  chunks.push(code`
+    decodeDelimited(
+      input: ${Reader} | Uint8Array,
+    ): ${fullName} {
+      const reader = input instanceof ${Reader} ? input : new ${Reader}(input);
+      let end = reader.uint32();
```

Comment:
Instead of repeating the entire contents of `decode`, could this be something like:

```typescript
const reader = ...
let end = reader.uint32();
return decode(reader, end);
```

I.e. `decode` already accepts an optional `length` argument, which I think? is the only thing that has changed here?

Reply:
You're right! Since `delimitedMethods` depends on `outputEncodeMethods` we can definitely reuse that logic.